test/src/server/helpers/revisions.js

```diff
@@ -0,0 +1,415 @@
+		expect(orderedRevisions).to.be.descendingBy('createdAt');
+	});
+
+	it('should return the expected subset of revisions when passed an offset (from)', async () => {
+		const from = 900;
+		const size = 10;
+		const allRevisions = await getOrderedRevisions(0, 1000, orm);
+		const orderedRevisions = await getOrderedRevisions(from, size, orm);
+		const allRevisionsSubset = allRevisions.slice(from, from + size);
+		expect(orderedRevisions.length).to.equal(size);
+		expect(orderedRevisions).to.deep.equal(allRevisionsSubset);
+		expect(orderedRevisions).to.be.descendingBy('createdAt');
+	});
+
+	it('should return no results if offset is higher than total revisions', async () => {
+		// only 1000 revisions were created
+		const orderedRevisions = await getOrderedRevisions(1001, 10, orm);
+		expect(orderedRevisions.length).to.be.equal(0);
+	});
+});
+
+describe('getAssociatedEntityRevisions', () => {
+	// eslint-disable-next-line one-var
+	let aliasSetId, authorJSON, editionGroupJSON, editionJSON,
+		editorJSON, expectedDefaultAliasId, publisherJSON, workJSON;
+	beforeEach(async () => {
+		const author = await createAuthor();
+		const edition = await createEdition();
+		const editionGroup = await createEditionGroup();
+		const publisher = await createPublisher();
+		const work = await createWork();
+		const editor = await createEditor();
+		authorJSON = author.toJSON();
+		editionJSON = edition.toJSON();
+		editorJSON = editor.toJSON();
+		editionGroupJSON = editionGroup.toJSON();
+		publisherJSON = publisher.toJSON();
+		workJSON = work.toJSON();
+
+		// In each revision, we will change the alias of entity to AuthorJSON alias.
+		// We need expectedDefaultAliasId to check the test
+		const aliasSet = await new AliasSet().where({id: authorJSON.aliasSetId}).fetch();
+		const aliasSetJSON = aliasSet.toJSON();
+		aliasSetId = aliasSetJSON.id;
+		expectedDefaultAliasId = aliasSetJSON.defaultAliasId;
+	});
+	afterEach(truncateEntities);
+
+	it('should return formatted revisions array after adding entity (1 revision with 1 entity revised)', async () => {
+		// here author is revised
+		const revisionId = random.number();
+		await new Revision({authorId: editorJSON.id, id: revisionId})
+			.save(null, {method: 'insert'});
+		const dataId = random.number();
+		// we can ignore other fields expect aliasSetId. We are interested in fetching alias only
+		await new AuthorData({aliasSetId, id: dataId}).save(null, {method: 'insert'});
+		await new AuthorRevision({bbid: authorJSON.bbid, dataId, id: revisionId}).save(null, {method: 'insert'});
+
+		const formattedRevision = [
+			{
+				authorId: editorJSON.id,
+				entities: [],
+				revisionId
+			}
+		];
+		const revisionsWithEntities = await getAssociatedEntityRevisions(formattedRevision, orm);
+
+		expect(revisionsWithEntities.length).to.be.equal(1);
+		expect(revisionsWithEntities[0].entities.length).to.be.equal(1);
+		revisionsWithEntities[0].entities.forEach((entity) => {
```

Comment:
Instead of a `forEach` here, considering you test the length just above you can use `revisionsWithEntities[0].entities[0].defaultAlias.id` which is a bit clearer.